src/test/regress/input/multi_shard_modify.source

```diff
@@ -0,0 +1,135 @@
+
+-- multi-column UPDATE
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_comment=''Citus Rocks!'', c_address=''SF'' WHERE c_custkey>998 AND c_custkey<1002');
+SELECT c_comment, c_address FROM customer_delete_protocol WHERE c_custkey>998 AND c_custkey<1002;
+
+-- commands with no constraints on the partition key are supported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_mktsegment=''Citus'' WHERE c_mktsegment=''MACHINERY'' ');
+SELECT count(*) FROM customer_delete_protocol WHERE c_mktsegment='Citus';
+
+-- attempting to change the partition key is unsupported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_custkey=3000 WHERE c_custkey < 3000 ');
+
+-- UPDATEs with a FROM clause are unsupported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_comment = ''FAIL'' FROM temp_nations WHERE customer_delete_protocol.c_custkey = 777 AND customer_delete_protocol.c_nationkey = temp_nations.key AND temp_nations.name = ''dummy'' ');
+
+-- commands with a RETURNING clause are unsupported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_name=''Citus Data Customer'' WHERE c_custkey=353 RETURNING *');
+
+-- commands containing a CTE are unsupported
+SELECT master_modify_multiple_shards('WITH t AS (INSERT INTO customer_delete_protocol DEFAULT VALUES RETURNING *) UPDATE customer_delete_protocol SET c_comment = ''FAIL'' ');
+
+-- updates referencing just a var are supported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_nationkey=c_custkey WHERE c_custkey = 666');
+SELECT c_nationkey FROM customer_delete_protocol WHERE c_custkey=666;
+
+-- updates referencing a column are supported
+SELECT c_nationkey FROM customer_delete_protocol WHERE c_custkey = 678;
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_nationkey = c_nationkey + 37 WHERE c_custkey = 678');
+SELECT c_nationkey FROM customer_delete_protocol WHERE c_custkey = 678;
+
+-- updates referencing non-IMMUTABLE functions are unsupported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_comment = ''FAIL!'' WHERE c_custkey = temp_stable_func()');
+
+-- updates referencing IMMUTABLE functions in SET section are supported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_nationkey = abs(-78) WHERE c_custkey = 777');
+SELECT c_nationkey FROM customer_delete_protocol WHERE c_custkey = 777;
+
+-- updates referencing STABLE functions in SET section are not supported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_nationkey = temp_stable_func() * 2 WHERE c_custkey = 888');
+
+-- updates referencing VOLATILE functions in SET section are not supported
+SELECT master_modify_multiple_shards('UPDATE customer_delete_protocol SET c_nationkey = random() WHERE c_custkey = 888');
+
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 102046;
```

Comment:
Minor: GitHub/Linux likes when there's an end-of-line at the end of a text file.